internal/controller/pause.go

```diff
@@ -19,110 +19,59 @@ var (
 
 func GetPauseModule() *PauseModule {
 	once.Do(func() {
-		pauseModuleInstance = &PauseModule{
-			controllerRequestedPause: make(map[string]*bool),
-			isbSvcRequestedPause:     make(map[string]*bool),
-		}
+		pauseModuleInstance = &PauseModule{pauseRequests: make(map[string]*bool)}
 	})
 
 	return pauseModuleInstance
 }
 
 type PauseModule struct {
-	// map of namespace (where Numaflow Controller lives) to Pause Request
-	controllerRequestedPause     map[string]*bool // having *bool gives us 3 states: [true=pause-required, false=pause-not-required, nil=unknown]
-	controllerRequestedPauseLock sync.RWMutex
-	// map of ISBService namespaced name to Pause Request
-	isbSvcRequestedPause     map[string]*bool // having *bool gives us 3 states: [true=pause-required, false=pause-not-required, nil=unknown]
-	isbSvcRequestedPauseLock sync.RWMutex
+	lock sync.RWMutex
+	// map of pause requester to Pause Request
+	pauseRequests map[string]*bool // having *bool gives us 3 states: [true=pause-required, false=pause-not-required, nil=unknown]
```

Comment:
simplified all of this code into a single set of `pauseRequests` which are mapped by uniquely named "pause requesters"